operator/cmd/syncclusterconfig/sync.go

```diff
@@ -234,141 +239,184 @@ type ClusterConfigStatus struct {
 	status, err = s.Client.ClusterConfigStatus(ctx, true)
 	if err != nil {
 		return nil, err
 	}
+
+	needsRestart := false
+	for _, s := range status {
+		configVersion = max(s.ConfigVersion, configVersion)
+		needsRestart = needsRestart && s.Restart
+	}
+
 	return &ClusterConfigStatus{
-		Version:                       int64(result.ConfigVersion),
-		NeedsRestart:                  slices.ContainsFunc(status, func(s rpadmin.ConfigStatus) bool { return s.Restart }),
+		Version:                       configVersion,
+		NeedsRestart:                  needsRestart,
 		PropertiesThatNeedRestartHash: hashOfConfigsThatNeedRestart,
 	}, nil
 }
 
-func (s *Syncer) maybeMergeSuperusers(ctx context.Context, config map[string]any, superusers []string) {
+func (s *Syncer) normalizeConfig(
+	ctx context.Context,
+	schema rpadmin.ConfigSchema,
+	superusers []string,
+	desired map[string]any,
+) map[string]any {
 	logger := log.FromContext(ctx)
+	normalized := maps.Clone(desired)
+
+	if superusers, ok := s.mergeSuperusers(ctx, superusers, desired); ok {
+		// Conversely, if ok is false, we don't need to unset superusers as it
+		// should already be unset.
+		normalized["superusers"] = superusers
+	}
 
-	superusersConfig, ok := config[superusersEntry]
-	if !ok {
-		// we have no superusers configuration, so don't do anything
-		logger.Info("Configuration does not contain a 'superusers' entry. Skipping superusers merge.")
-		return
+	// Redpanda's cluster configs can be aliased. The preferred name is the one
+	// that's listed in the schema (as opposed to being present in an alias
+	// list).
+	for key, schema := range schema {
+		_, hasKey := normalized[key]
+
+		for _, alias := range schema.Aliases {
+			aliased, hasAlias := normalized[alias]
+
+			if !hasKey && hasAlias {
+				logger.Info("renaming aliased config key", "from", alias, "to", key)
+				normalized[key] = aliased
+				delete(normalized, alias)
+			} else if hasKey && hasAlias {
+				logger.Info("dropping aliased key due to non-alias key being provided", "alias", alias, "non-alias", key)
+				delete(normalized, alias)
+			}
+		}
 	}
 
-	switch su := superusersConfig.(type) {
-	case []string:
-		superusers = append(superusers, su...)
+	return normalized
+}
+
+func (s *Syncer) mergeSuperusers(ctx context.Context, superusers []string, desired map[string]any) ([]string, bool) {
+	desiredSUs, ok := desired["superusers"]
+	superusers = append(superusers, coherceToSliceOf[string](ctx, desiredSUs)...)
+
+	// return union(desired['superusers'], superusers) and an indication of
+	// whether or not the superusers key should be set or not.
+	//
+	// **Setting the key or not has different implications depending on s.Mode**
+	//
+	// We largely key off the presence, or lack thereof, of the superusers key
+	// in the desired cluster config. If any superusers are provided via other
+	// means, we consider that as an implicit setting of superusers.
+	return NormalizeSuperusers(superusers), ok || len(superusers) > 0
```

Comment:
Prior to this change the second return here was treated as `ok && len(superusers) > 0`